libraries/AP_Scheduler/AP_Scheduler.cpp

```diff
@@ -89,96 +90,87 @@ void AP_Scheduler::tick(void)
  */
 void AP_Scheduler::run(uint32_t time_available)
 {
-    uint32_t run_started_usec = AP_HAL::micros();
-    uint32_t now = run_started_usec;
-
     if (_debug > 1 && _perf_counters == nullptr) {
         _perf_counters = new AP_HAL::Util::perf_counter_t[_num_tasks];
-        if (_perf_counters != nullptr) {
```

Comment:
why remove this check ? it is because if it didn't manage to allocate the perf_counter , we are already in bad shape ?

Reply:
ok, after checking we need to keep the check, as for example SITL doesn't implement the perf_counter, so it return a nullptr !

Reply:
How can in SITL the call of "new" return a nullptr?

Reply:
my bad ! that perf_alloc that return the nullptr :-S